plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/kernel/clients/KernelParquetHandler.java

```diff
@@ -0,0 +1,214 @@
+import io.delta.kernel.types.StructType;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.kernel.utils.FileStatus;
+import io.trino.filesystem.Location;
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.metastore.HiveType;
+import io.trino.parquet.ParquetReaderOptions;
+import io.trino.plugin.deltalake.DeltaHiveTypeTranslator;
+import io.trino.plugin.deltalake.kernel.KernelSchemaUtils;
+import io.trino.plugin.deltalake.kernel.data.TrinoColumnarBatchWrapper;
+import io.trino.plugin.hive.FileFormatDataSourceStats;
+import io.trino.plugin.hive.HiveColumnHandle;
+import io.trino.plugin.hive.ReaderPageSource;
+import io.trino.plugin.hive.parquet.ParquetPageSourceFactory;
+import io.trino.spi.Page;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeManager;
+import org.apache.hadoop.conf.Configuration;
+import org.joda.time.DateTimeZone;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalLong;
+
+import static io.trino.plugin.hive.parquet.ParquetPageSourceFactory.PARQUET_ROW_INDEX_COLUMN;
+import static java.util.Objects.requireNonNull;
+
+public class KernelParquetHandler
+        implements ParquetHandler
+{
+    private final Configuration configuration;
+    private final TrinoFileSystem fileSystem;
+    private final TypeManager typeManager;
+
+    public KernelParquetHandler(
+            Configuration configuration,
+            TrinoFileSystem fileSystem,
+            TypeManager typeManager)
+    {
+        this.configuration = configuration;
+        this.fileSystem = fileSystem;
+        this.typeManager = typeManager;
+    }
+
+    @Override
+    public CloseableIterator<ColumnarBatch> readParquetFiles(
+            CloseableIterator<FileStatus> closeableIterator,
+            StructType structType,
+            Optional<Predicate> optional)
+            throws IOException
+    {
+        return readFiles(closeableIterator, structType);
+    }
+
+    private CloseableIterator<ColumnarBatch> readFiles(CloseableIterator<FileStatus> fileIter, StructType physicalSchema)
+    {
+        return new CloseableIterator<>()
+        {
+            private CloseableParquetBatchReader currentFileReader;
+
+            @Override
+            public void close()
+            {
+                Utils.closeCloseables(fileIter, currentFileReader);
+            }
+
+            @Override
+            public boolean hasNext()
+            {
+                if (currentFileReader != null && currentFileReader.hasNext()) {
+                    return true;
+                }
+
+                Utils.closeCloseables(currentFileReader);
+                if (!fileIter.hasNext()) {
```

Comment:
nit: `fileIter` -> `fileIterator`

Trino code conventions state that variable name shortcuts are disincouraged.